### PR TITLE
Add autoscalingClusterConfiguration test

### PR DESCRIPTION
**Pull request by maryvictol**
This PR adds testcase and e2e-test for Autoscaling cluster configuration.

### CODE REVIEW

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 1]
```diff
@@ -98,11 +109,22 @@ public class LaunchParametersTest extends AbstractSeveralPipelineRunningTest
     private final String registry = C.DEFAULT_REGISTRY;
     private final String group = C.DEFAULT_GROUP;
     private final String tool = C.TESTING_TOOL_NAME;
+    private final String anotherGroup = C.ANOTHER_GROUP;
+    private final String testTool = format("%s/%s", anotherGroup, C.TEST_DOCKER_IMAGE);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Unused field

**Review by NShaforostov — APPROVED** [review]

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 2]
```diff
@@ -98,11 +109,22 @@ public class LaunchParametersTest extends AbstractSeveralPipelineRunningTest
...
     private final String customTag = "test_tag";
     private static String testInstance = "r6i.%s";
     private int[] scaling = new int[4];
     private int[] sizeDisk = new int[4];
     private String initialLaunchSystemParameters;
+    private String testProfile = "testprofile.q";
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Field may be 'final'

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 3]
```diff
@@ -518,6 +540,81 @@ public void addSupportForAutoscalingFilesystemForAWS() {
...
+                .perform(registry, group, tool, ToolTab::runWithCustomSettings)
+                .expandTab(EXEC_ENVIRONMENT)
+                .expandTab(ADVANCED_PANEL)
+                .setPriceType(ON_DEMAND)
+                .doNotMountStoragesSelect(true)
+                .enableClusterLaunch()
+                .clusterSettingsForm("Auto-scaled cluster")
+                .setWorkingNodesCount("1")
+                .ok()
+                .launchTool(this, nameWithoutGroup(tool));//
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Remove the empty slashes, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 4]
```diff
@@ -518,6 +540,81 @@ public void addSupportForAutoscalingFilesystemForAWS() {
...
+                            .execute(command[2])
+                            .assertPageContains(format("Grid engine %s profile.", testProfile))
+                            .getRowsNumber();
+                    shell
+                            .execute(format("%sdd", num))
+                            .inputText(command[3])
+                            .execute(":wq")
+                            .waitUntilTextAppears(parentRunId)
+                            .assertPageContainsString("Boolean parameter CP_CAP_AUTOSCALE has invalid value nontrue. Please specify true/false/yes/no/on/off.")
+                            .assertPageContainsString("Grid engine profile verification has failed. Reverting the changes...")
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Add line breaks, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 5]
```diff
@@ -518,6 +540,81 @@ public void addSupportForAutoscalingFilesystemForAWS() {
...
+                            .close();
+                });
+
+        Set<String> logMessages = runsMenu()
+                .showLog(parentRunId)
+                .clickTaskWithName("SGEProfiles")
+                .logMessages()
+                .collect(toSet());
+        Arrays.stream(expectedLogMessages())
+                .map(String::trim)
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: I suppose we can avoid using `map(String::trim)` by performing the trimming inside the forEach method, isn't it?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java [hunk 6]
```diff
@@ -594,4 +691,12 @@ private void checkFilesystemSpace(ShellAO shell, int diskSize) {
                 .assertNextStringIsVisible(CHECK_SPACE_COMMAND, rootHost)
                 .assertPageAfterCommandContainsStrings(CHECK_SPACE_COMMAND, logMessage(diskSize));
     }
+
+    private String[] expectedLogMessages() {
+        List<String> loglist = new ArrayList<String>();
+        loglist.addAll(asList(command[4].replace("I ", "")
+                .replaceAll("export", "> export").split("\n")));
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: 'addAll()' call can be replaced with parametrized constructor call

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/ShellAO.java [hunk 7]
```diff
@@ -85,6 +85,14 @@ public ShellAO execute(String command) {
...
     }
 
+    public ShellAO inputText(String command) {
+        sleep(300, MILLISECONDS);
+        Utils.sendKeysWithSlashes(command);
+        sleep(300, MILLISECONDS);
+        actions().sendKeys(Keys.ESCAPE).perform();
+        return this;
+    }
+
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/ShellAO.java]: We already have the similar method `execute()`. Let's reuse that - create common method `execute(String command, CharSequence key)`, and inputText() will be execute(command, Keys.ESCAPE)

File: e2e/gui/src/test/resources/parameters_file.txt
Comment (kamyshova) [inline e2e/gui/src/test/resources/parameters_file.txt:1]: Let's rename this file to be more specific with the test

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 1 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 2 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 3 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 4 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/ShellAO.java (hunk 7 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/ShellAO.java]: Done

File: e2e/gui/src/test/resources/parameters_file.txt
Comment (maryvictol) [reply e2e/gui/src/test/resources/parameters_file.txt:1]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 5 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java (hunk 6 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchParametersTest.java]: Done

**Review by kamyshova — APPROVED** [review]

**Pull request merged**